modular/gater/object_handler.go

```diff
@@ -123,6 +123,69 @@ func parseRange(rangeStr string) (bool, int64, int64) {
 
 // resumablePutObjectHandler handles the resumable put object
 func (g *GateModular) resumablePutObjectHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		err        error
+		reqCtx     *RequestContext
+		authorized bool
+		objectInfo *storagetypes.ObjectInfo
+		params     *storagetypes.Params
+	)
+
+	defer func() {
+		reqCtx.Cancel()
+		if err != nil {
+			reqCtx.SetError(gfsperrors.MakeGfSpError(err))
+			reqCtx.SetHttpCode(int(gfsperrors.MakeGfSpError(err).GetHttpStatusCode()))
+			MakeErrorResponse(w, gfsperrors.MakeGfSpError(err))
+		} else {
+			reqCtx.SetHttpCode(http.StatusOK)
+		}
+		log.CtxDebugw(reqCtx.Context(), reqCtx.String())
+	}()
+
+	reqCtx, err = NewRequestContext(r, g)
+	if err != nil {
+		return
+	}
+	if reqCtx.NeedVerifyAuthorizer() {
+		authorized, err = g.baseApp.GfSpClient().VerifyAuthorize(reqCtx.Context(),
+			coremodule.AuthOpTypePutObject, reqCtx.Account(), reqCtx.bucketName, reqCtx.objectName)
+		if err != nil {
+			log.CtxErrorw(reqCtx.Context(), "failed to verify authorize", "error", err)
+			return
+		}
+		if !authorized {
+			log.CtxErrorw(reqCtx.Context(), "no permission to operate")
+			err = ErrNoPermission
+			return
+		}
+	}
+
+	objectInfo, err = g.baseApp.Consensus().QueryObjectInfo(reqCtx.Context(), reqCtx.bucketName, reqCtx.objectName)
+	if err != nil {
+		log.CtxErrorw(reqCtx.Context(), "failed to get object info from consensus", "error", err)
+		err = ErrConsensus
+		return
+	}
+	if objectInfo.GetPayloadSize() == 0 || objectInfo.GetPayloadSize() > g.maxPayloadSize {
```

Comment:
It's necessary to check the offset and complete parameters. E.g. 
- whether the offset equal with uploaded offset 
- whether the payload size is a multiple of the segment size, expect the last part.

Reply:
There is no need to check `MaxPayloadSize` here, as it has already been validated during the CreateObject process. If the object creation was successful, the storage provider should accept the `PutObject` request.